core/src/main/scala/io/finch/ToService.scala

```diff
@@ -102,42 +101,23 @@ object ToService {
   implicit def hlistTS[F[_], A, EH <: Endpoint[F, A], ET <: HList, CTH, CTT <: HList](implicit
     ntrA: ToResponse.Negotiable[F, A, CTH],
     ntrE: ToResponse.Negotiable[F, Exception, CTH],
-    F: Effect[F],
-    tsT: ToService[ET, CTT],
+    F: MonadError[F, Throwable],
+    tsT: ToService[F, ET, CTT],
     isNegotiable: IsNegotiable[CTH]
-  ): ToService[Endpoint[F, A] :: ET, CTH :: CTT] = new ToService[Endpoint[F, A] :: ET, CTH :: CTT] {
-    def apply(es: Endpoint[F, A] :: ET, opts: Options, ctx: Context): Service[Request, Response] =
-      new Service[Request, Response] {
+  ): ToService[F, Endpoint[F, A] :: ET, CTH :: CTT] = new ToService[F, Endpoint[F, A] :: ET, CTH :: CTT] {
+    def apply(es: Endpoint[F, A] :: ET, opts: Options, ctx: Context): Service[F] =
+      new Service[F] {
         private[this] val handler =
           if (opts.enableUnsupportedMediaType) respond415.orElse(respond400) else respond400
 
         private[this] val negotiateContent = isNegotiable.fold(_ => true, _ => false)
         private[this] val underlying = es.head.handle(handler)
 
-        def apply(req: Request): Future[Response] = underlying(Input.fromRequest(req)) match {
+        def apply(req: Request): F[(Trace, Response)] = underlying(Input.fromRequest(req)) match {
           case EndpointResult.Matched(rem, trc, out) if rem.route.isEmpty =>
 
-            Trace.captureIfNeeded(trc)
```

Comment:
This is backward-incompatible also. Should we stick to propagate it both through the context and return type?